server/src/main/java/ch/ethz/lapis/api/SampleService.java

```diff
@@ -388,6 +406,28 @@ public void getFasta(
         }
     }
 
+    private  <SequenceRecord extends Record> void streamFastaSequences(
+        OrderAndLimitConfig orderAndLimit,
+        OutputStream outputStream,
+        TableField<SequenceRecord, byte[]> sequenceColumn,
+        TableField<YMainMetadataRecord, String> sequenceIdentifierColumn,
+        SelectJoinStep<Record2<String, byte[]>> statement
+    ) {
+        if (orderAndLimit.getLimit() == null) {
+            orderAndLimit.setLimit(100000);
+        }
+        var orderedAndLimitedstatement = applyOrderAndLimit(statement, orderAndLimit);
+
+        try (var cursor = orderedAndLimitedstatement.fetchSize(1000).fetchLazy()) {
+            for (var row : cursor) {
+                var fastaEntry = new FastaEntry(
+                    row.get(sequenceIdentifierColumn),
+                    referenceSeqCompressor.decompress(row.get(sequenceColumn))
```

Comment:
The `referenceSeqCompressor` as defined above uses the dictionary `ZstdSeqCompressor.DICT.REFERENCE`. This is however only for nucleotide sequences. AA sequences are compressed `ZstdSeqCompressor.DICT.AA_REFERENCE` ([source](https://github.com/GenSpectrum/LAPIS/blob/3a975980e83b781e1ccf03f6773d2885934f7406/server/src/main/java/ch/ethz/lapis/source/gisaid/BatchProcessingWorker.java#L156)).

(Sorry for all the non-documented details!)